services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3CrtAsyncHttpClient.java

```diff
@@ -130,6 +134,13 @@ public CompletableFuture<Void> execute(AsyncExecuteRequest asyncRequest) {
             .withResponseHandler(responseHandler)
             .withResumeToken(resumeToken);
 
+        // Create a new SigningConfig object only if the signing region has changed from the previously configured region.
+        if (signingRegion != null && !s3ClientOptions.getRegion().equals(signingRegion)) {
```

Comment:
`s3ClientOptions.getRegion().equals(signingRegion)` is comparing different types, `String` and `Region`. Should be using `signingRegion.id()`

Reply:
Good catch !